pkg/controllers/managementuser/rbac/globalrolebinding_handler.go

```diff
@@ -79,59 +77,48 @@ func newGlobalRoleBindingHandler(workload *config.UserContext) v3.GlobalRoleBind
 // grbHandler ensures the global admins have full access to every cluster. If a globalRoleBinding is created that uses
 // the admin role, then the user in that binding gets a clusterRoleBinding in every user cluster to the cluster-admin role
 type grbHandler struct {
-	clusterName                 string
-	clusterRoleBindings         rbacv1.ClusterRoleBindingInterface
-	crbLister                   rbacv1.ClusterRoleBindingLister
-	grbIndexer                  cache.Indexer
-	grLister                    v3.GlobalRoleLister
-	rbLister                    rbacv1.RoleBindingLister
-	roleBindings                rbacv1.RoleBindingInterface
-	globalroleBindingController v3.GlobalRoleBindingController
-	clusters                    v3.ClusterInterface
-	provClusters                provisioningcontrollers.ClusterCache
+	clusterName         string
+	clusterRoleBindings rbacv1.ClusterRoleBindingInterface
+	crbLister           rbacv1.ClusterRoleBindingLister
+	grbCache            mgmtv3.GlobalRoleBindingCache
+	crtbCache           mgmtv3.ClusterRoleTemplateBindingCache
+	crtbCtrl            mgmtv3.ClusterRoleTemplateBindingClient
+	grLister            v3.GlobalRoleLister
 }
 
-func (c *grbHandler) sync(key string, obj *v3.GlobalRoleBinding) (runtime.Object, error) {
+func (c *grbHandler) sync(key string, obj *apisv3.GlobalRoleBinding) (runtime.Object, error) {
 	if obj == nil || obj.DeletionTimestamp != nil {
 		return obj, nil
 	}
 
+	if obj.GlobalRoleName == rbac.GlobalRestrictedAdmin {
+		// restricted admin only needs to be get a CRTB to cluster-owner RoleTemplate
+		return obj, c.ensureClusterOwnerTemplateBinding(obj)
+	}
+
 	isAdmin, err := c.isAdminRole(obj.GlobalRoleName)
 	if err != nil {
 		return nil, err
-	} else if !isAdmin {
-		return obj, nil
 	}
-
-	// Do not sync restricted-admin to the local cluster as 'cluster-admin'
-	if c.clusterName == "local" && obj.GlobalRoleName == rbac.GlobalRestrictedAdmin {
+	if !isAdmin {
 		return obj, nil
 	}
 
 	logrus.Debugf("%v is an admin role", obj.GlobalRoleName)
 
-	err = c.ensureClusterAdminBinding(obj)
-	if err != nil {
-		return nil, err
-	}
-
-	err = c.ensureProvisioningClusterAdminBinding(obj)
-	if err != nil {
-		return nil, err
-	}
-
-	return obj, nil
+	return obj, c.ensureClusterAdminBinding(obj)
 }
 
-// ensureClusterAdminBinding creates a cluster-admin binding in the downstream cluster
-func (c *grbHandler) ensureClusterAdminBinding(obj *v3.GlobalRoleBinding) error {
+// ensureAdminAsClusterAdmin creates a ClusterRoleBinding for GRB subject to
```

Comment:
```suggestion
// ensureClusterAdminBinding creates a ClusterRoleBinding for GRB subject to
```
or make the method name match

Reply:
I will update the comment. I ended up reverting the name to its original form.

Reply:
fixed.